### ISSUE TITLE
Check the methods of types

### ISSUE THREAD

**Issue by Feufochmar** [RobLoach/lutro-status#3]
Currently, the script checks if modules and functions are present. However, since Löve2D is object-oriented, the functions defined inside the tables returned by the newXXX functions are also part of the API. (or in other words, the methods on the objects returned by the constuctor functions).
Currently, the script does not check if methods are present. This requires work as it is needed to instanciate a type to check if its methods are present or missing.

When checking a type, the script should:
 1) check if the type has a constructor
 2) check the constructor definition (to check for a suitable variant to call)
 3) call the constructor
 4) check if the methods are present
 5) check if the methods of the supertypes are present

Since a constructor may need other objects to be called, it may not be possible to do the 3rd step automatically  from the api description for all types. Those types could be checked with manual code for the instanciation.
The 5th step requires to build a dictionnary of the types defined by the API, as the documentation is written as a tree and not as a map.

**Comment by RobLoach** [comment]
Agreed, [`love.joystick.getJoysticks`](https://love2d.org/wiki/love.joystick.getJoysticks) is an example of bringing in a Lua object.

Kivutar mentioned there are a few examples of returning objects in Lutro, but I haven't looked further into it.
